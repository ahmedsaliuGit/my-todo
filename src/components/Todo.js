import React from "react";

import "./components.css";
import { limitText } from "../helpers";

const Todo = (props) => {
  const { item, handleClick } = props;

  return (
    <>
      <li>
        <span className="todo-item">{limitText(item.name)}</span>
        <span
          className="todo-item-complete"
          style={{
            backgroundColor: !item.complete ? "red" : "green",
            color: !item.complete ? "white" : "default",
          }}
          onClick={handleClick}
        >
          {item.complete ? "Completed" : "Not complete"}
        </span>
      </li>
    </>
  );
};

export default Todo;
