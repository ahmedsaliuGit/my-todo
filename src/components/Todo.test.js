import React from "react";
import renderer from "react-test-renderer";

import Todo from "./Todo";

test("Todo Component", () => {
  const itemObj = {
    id: Date.now(),
    name: "Some Todo",
    complete: false,
  };
  const component = renderer.create(
    <Todo
      key="http://www.facebook.com"
      item={itemObj}
      handleClick={() => console.log()}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
