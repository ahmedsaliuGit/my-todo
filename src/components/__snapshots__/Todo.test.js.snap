// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component 1`] = `
<li>
  <span
    className="todo-item"
  >
    Some Todo
  </span>
  <span
    className="todo-item-complete"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "color": "white",
      }
    }
  >
    Not complete
  </span>
</li>
`;
